x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_schemas.ts

```diff
@@ -109,71 +122,86 @@ export const baseRuleParams = t.exact(
 );
 export type BaseRuleParams = t.TypeOf<typeof baseRuleParams>;
 
-const eqlSpecificRuleParams = t.type({
-  type: t.literal('eql'),
-  language: t.literal('eql'),
-  index: indexOrUndefined,
-  query,
-  filters: filtersOrUndefined,
-  eventCategoryOverride: eventCategoryOverrideOrUndefined,
-});
+const eqlSpecificRuleParams = t.intersection([
+  t.type({
+    type: t.literal('eql'),
+    language: t.literal('eql'),
+    index: indexOrUndefined,
+    query,
+    filters: filtersOrUndefined,
+    eventCategoryOverride: eventCategoryOverrideOrUndefined,
+  }),
+  t.partial({ dataViewId: dataViewIdOrUndefined }),
+]);
```

Comment:
Yup this makes sense. I think when I was testing out import / export I needed to make dataViewId partial but that is not a problem because the import_rules_schema handles that, not the rule type schemas.

Reply:
resolved here: [`2ce7c44` (#130929)](https://github.com/elastic/kibana/pull/130929/commits/2ce7c440573964a598730a124478cb7ed29d5d88)